LML.NPOManagement.Bll/Services/UserService.cs

```diff
@@ -263,5 +263,44 @@ public async Task<bool> ModifyUserInfo(UserInformationModel userInformationModel
             await _dbContext.SaveChangesAsync();
             return true;
         }
+
+        public async Task<List<UserInformationModel>> GetUserByUsername(string userName, bool showGroupsOnly, int userId)
+        {
+            List<User> users = new();
+
+            if (!showGroupsOnly)
+            {
+                users = await _dbContext.Users.ToListAsync();
+            }
+            else
+            {
+                var groupIds = _dbContext.Users
+                    .Where(u => u.Id == userId)
+                    .SelectMany(u => u.Groups.Select(g => g.Id))
+                    .ToList();
+
+                users = _dbContext.Users
+                    .Where(u => u.UsersGroups.Any(g => groupIds.Contains(g.Id)))
+                    .ToList();
```

Comment:
Aren't group ids the PrimaryKey of the UserGroup table? We should have userId and userName properties, please verify with DB model that you are using the right field